x-pack/plugins/cloud_security_posture/server/routes/benchmark_rules/bulk_action/utils.ts

```diff
@@ -29,14 +111,63 @@ export const updateRulesStates = async (
   );
 };
 
-export const setRulesStates = (ruleIds: string[], state: boolean): CspBenchmarkRulesStates => {
+export const setRulesStates = (
+  ruleIds: string[],
+  state: boolean,
+  benchmarkRules: CspBenchmarkRule[]
+): CspBenchmarkRulesStates => {
   const rulesStates: CspBenchmarkRulesStates = {};
-  ruleIds.forEach((ruleId) => {
-    rulesStates[ruleId] = { muted: state };
+  ruleIds.forEach((ruleId, index) => {
+    const benchmarkRule = benchmarkRules[index];
+    rulesStates[ruleId] = {
+      muted: state,
+      benchmark_id: benchmarkRule.metadata.benchmark.id,
+      benchmark_version: benchmarkRule.metadata.benchmark.version,
+      rule_number: benchmarkRule.metadata.benchmark.rule_number
+        ? benchmarkRule.metadata.benchmark.rule_number
+        : '',
```

Comment:
1. is `rule_number` an actual number or a string number like '1.2.3'? if its an actual number than defaulting to an empty string can be misleading.
2. do you think we can use `undefined` instead of an empty string?
3. you can do `benchmarkRule.metadata.benchmark.rule_number || ''` which means the same thing that you did but shorter.

Reply:
1. It's a string.
2. I am not sure since the interface `CspBenchmarkRulesStates` is defined using `schema` which as far as I know, does not support undefined.
3. Done